plugins/woocommerce/.claude/CLAUDE.md

```diff
@@ -0,0 +1,145 @@
+- When modifying existing code, if the git diff shows changes to a line that fires a hook without a docblock, add a docblock for that hook. Use `git log` to determine which version introduced the hook, and add the appropriate `@since` annotation with that version number.
+
+**Unit test file conventions:**
+
+- For classes in `includes/{path}/class-wc-{name}.php`, the test file is `tests/php/includes/{path}/class-wc-{name}-test.php`. The test class name is the same as the tested class with a `_Test` suffix.
+- For classes in `src/{path}/{name}.php`, the test file is `tests/php/src/{path}/{name}Test.php`.
+- When adding or modifying a unit test method, the part of the docblock that describes the test must be prepended with `@testdox`.
+- Test configuration file: `phpunit.xml`
+
+
+## Coding style
+
+- Prefer the null coalescing operator instead of executing `isset` before accessing an array item that might not exist. For example, use `if ( 34 === ( $foo['bar'] ?? null ) )` instead of `if ( isset( $foo['bar'] ) && 34 === $foo['bar'] )`.
+
+
+## Running Tests
+
+### PHP Unit Tests
+
+To run PHP unit tests in the WooCommerce plugin directory, use the following commands:
+
+```bash
+# Run all PHP unit tests
+pnpm run test:php:env
+
+# Run specific test class
+pnpm run test:php:env -- --filter TestClassName
+
+# Run specific test method
+pnpm run test:php:env -- --filter TestClassName::test_method_name
+
+# Run tests with verbose output
+pnpm run test:php:env -- --verbose --filter TestClassName
+
+# Examples:
+pnpm run test:php:env -- --filter PaymentsExtensionSuggestionsTest
+pnpm run test:php:env -- --filter PaymentsExtensionSuggestionsTest::test_get_country_extensions_count_with_merchant_selling_online
+```
+
+### Test Environment
+
+- Tests run in a Docker-based WordPress environment using `wp-env`
+- The test environment is automatically set up and configured
+- Tests use PHPUnit 9.6.24 with PHP 8.0.30
+- WordPress and WooCommerce are automatically installed in the test environment
+
+
+### Common Test Commands
+
+```bash
+# Run tests for a specific directory
+pnpm run test:php:env -- tests/php/src/Internal/Admin/
+
+# Run tests matching a pattern
+pnpm run test:php:env -- --filter "Admin.*Test"
+
+# Run tests and stop on first failure
+pnpm run test:php:env -- --stop-on-failure
+
+# Get test coverage (if configured)
+pnpm run test:php:env -- --coverage-text
+```
+
+### Troubleshooting Tests
+
+- **Tests failing due to missing dependencies**: Ensure `npm install` has been run
+- **Docker issues**: Try `wp-env start` to restart the test environment
+- **Permission issues**: Tests run in Docker containers with proper permissions
+- **Xdebug warnings**: These can be safely ignored - they don't affect test results
+
+
+## Code Quality Commands
+
+When making changes to the codebase, run these commands to ensure code quality:
+
+```bash
+# Run PHP linting
+pnpm run lint:php
```

Comment:
Suggestion here: I'd recommend using the `lint:changes:branch:php` and `lint:changes:branch:js` commands unless explicitly asked to use the main lint command by the user, because we have a lot of legacy files which have linting errors and this would result in lots of lint errors not relevant to the current PR being reported (those lint errors should also be fixed at some point ideally, but that seems out of scope here 😆)

Reply:
Good catch, updated.